crates/sargon/src/signing/petition_types/petition_for_factors_types/petition_for_factors/petition_for_factors_input.rs

```diff
@@ -67,30 +67,30 @@ impl PetitionForFactorsInput {
         self.factors.len() as i8
     }
 
-    fn remaining_factors_until_success(
+    fn remaining_factors_until_success<S: Signable>(
```

Comment:
I think better put `PhantomData<T>` , where `T: SignableID` as a field and NOT let all thos methods take a `S` which looks wrong.